presto-hive/src/test/java/com/facebook/presto/hive/TestHiveSplitSource.java

```diff
@@ -249,6 +253,69 @@ public void testEmptyBucket()
         assertEquals(getSplits(hiveSplitSource, OptionalInt.of(3), 10).size(), 0);
     }
 
+    @Test
+    public void testPreloadSplitsForGroupedExecution()
+            throws Exception
+    {
+        // TODO: Use Session::builder after HiveTestUtils::SESSION is refactored to Session.
+        ConnectorSession session = new TestingConnectorSession(
+                new HiveSessionProperties(
+                        new HiveClientConfig().setPreloadSplitsForGroupedExecution(true),
+                        new OrcFileWriterConfig(),
+                        new ParquetFileWriterConfig()).getSessionProperties());
+        HiveSplitSource hiveSplitSource = HiveSplitSource.bucketed(
+                session,
+                "database",
+                "table",
+                TupleDomain.all(),
+                10,
+                10,
+                new DataSize(1, MEGABYTE),
+                new TestingHiveSplitLoader(),
+                Executors.newFixedThreadPool(5),
```

Comment:
We are creating a pool of 5 threads in each test (5 * N threads in total when N tests run in parallel). We should use a pool of a smaller size and share across tests. You can fix this in a follow up PR if you want.

Reply:
Put this into second refactor commit.